### ISSUE TITLE
[Bug]: Support of ARM servers breaks backwards compatibly

### ISSUE THREAD

**Issue by hegerdes** [hetznercloud/terraform-provider-hcloud#655]
### What happened?

When using existing terraform code the the resource creation should still work and not break with newly added features.

But when using the cod€described below terraform crashes when applied.

The error is:
```txt
Error: more than one Image found for name debian-11
```

### What did you expect to happen?

Old code should be backwards compatible. The ARM Images should have unique names like `debian-11-arm-v7`

**Workaround:**
Upgrade to the latest terraform provider:
```hcl
    hcloud = {
      source = "hetznercloud/hcloud"
      version = "1.38.0"
    }
```

### Please provide a minimal working example

```hcl
terraform {
  required_providers {
    hcloud = {
      source = "hetznercloud/hcloud"
      version = "1.35.1"
    }
  }
}

provider "hcloud" {
  token = var.hcloud_token
}

variable "hcloud_token" {
  sensitive = true # Requires terraform >= 0.14
  description = "Hetzner API Key"
}

# Create a new workers
resource "hcloud_server" "worker_nodes" {
  name        = "worker-node-xxx"
  image       = "debian-11"
  server_type = "cx21"
  location    = "nbg1"
  public_net {
    ipv4_enabled = true
    ipv6_enabled = true
  }
}

```

**Comment by mysticaltech** [comment]
@apricote FYI Same here, we are having this issue and the new version does not work for us anymore, we cannot use snapshots as it gives an error like `Error: image 106765659 for architecture x86 not found`.

https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner/issues/714

**Comment by mysticaltech** [comment]
@LKaemmerling @apricote Even though the snapshot image is saved with the right architecture, `hcloud_server` cannot use it since the last release.

![ksnip_20230413-030217](https://user-images.githubusercontent.com/518555/231619137-dbbc81f6-8d42-469b-8419-4e5f0aaf6a00.png)

**Comment by mysticaltech** [comment]
The error being hit is here https://github.com/hetznercloud/terraform-provider-hcloud/blob/2654d819f147dbe8367089d2dcea98d1e89d5564/internal/server/resource.go#L286

And even when we change the hcloud_image data, to use `with_architecture`, it finds it, but `hcloud_server` does not accept it, it generates the same error!

![ksnip_20230413-030631](https://user-images.githubusercontent.com/518555/231619504-26016fb8-f335-4553-b234-9835f07b2248.png)

**Comment by mysticaltech** [comment]
We are using both `cpx11` and `cx21` to test, and the same error shows up!

**Comment by mysticaltech** [comment]
The problem has been explained by @lars-hagen here https://github.com/hetznercloud/terraform-provider-hcloud/commit/2654d819f147dbe8367089d2dcea98d1e89d5564#r108628064

**Comment by mysticaltech** [comment]
To reproduce, just create a snapshot with packer, and try to load via data `hcloud_image` and create server from it. 

Or just follow the readme from [kube-hetzner/terraform-hcloud-kube-hetzner](https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner).

Currently it will fail with the above `Error: image XXXXX for architecture x86 not found`.

**Comment by apricote** [comment]
On it, thanks for the detailed reports :100:

**Comment by apricote** [comment]
@mysticaltech Your issue is different from the one described in the issue description. I have created a new issue #656 for your bug.

**Comment by apricote** [comment]
Hey @hegerdes,

we checked that the API changes would not break any of our official integration, but missed the check in the Terraform provider that verifies that there is only one image returned for the name.

We discussed various options and deemed this is the one with the smallest impact. It wasn't possible to not break something with this feature. We are working on our processes to ensure that this happens in a well communicated manner with appropriate timeframes for future API changes.

If there is anything blocking you from updating your version of the Terraform provider, please let me know so I can work on unblocking you as soon as possible.
